ui/src/main/java/org/pentaho/di/ui/i18n/MessagesSourceCrawler.java

```diff
@@ -208,38 +231,28 @@ public boolean traverseDescendents( FileSelectInfo info ) throws Exception {
 
         @Override
         public boolean includeFile( FileSelectInfo info ) throws Exception {
-          return info.getFile().getName().getExtension().equals( "java" );
+          FileObject file = info.getFile();
+          FileName fileName = file.getName();
+
+          return file.isFile() && JAVA_EXTENSION.equals( fileName.getExtension() ) && !filesToAvoid
+            .contains( fileName.getBaseName() );
         }
       } );
 
+      // Look for keys in each of the found files...
       for ( FileObject javaFile : javaFiles ) {
-
-        /**
-         * We don't want the Messages.java files, there is nothing in there for us.
-         */
-        boolean skip = false;
-        for ( String filename : filesToAvoid ) {
-          if ( javaFile.getName().getBaseName().equals( filename ) ) {
-            skip = true;
-          }
-        }
-        if ( skip ) {
-          continue; // don't process this file.
-        }
-
-        // For each of these files we look for keys...
-        //
         lookForOccurrencesInFile( sourceDirectory, javaFile );
       }
     }
+  }
+
+  protected void crawlXmlFolders() throws Exception {
 
-    // Also search for keys in the XUL files...
-    //
     for ( SourceCrawlerXMLFolder xmlFolder : xmlFolders ) {
-      String[] xmlDirs = { xmlFolder.getFolder(), };
-      String[] xmlMasks = { xmlFolder.getWildcard(), };
-      String[] xmlReq = { "N", };
-      boolean[] xmlSubdirs = { true, }; // search sub-folders too
+      String[] xmlDirs = { xmlFolder.getFolder() };
+      String[] xmlMasks = { xmlFolder.getWildcard() };
+      String[] xmlReq = { N_STRING };
```

Comment:
Why are we declaring a constant for something that is only used once?

Reply:
This is how the code was, I only removed the extra comma that existed in every array definition (a valid but - IMHO - confusing java spec) and reformatted the lines.
But you're correct: at first I've inlined them all but when I saw that the code readability was somewhat reduced, I've reverted my change and left it as it was.